pkg/operation/botanist/monitoring.go

```diff
@@ -60,7 +60,19 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 		usersDashboards     = strings.Builder{}
 	)
 
-	// Find extensions provider-specific monitoring configuration
+	// Fetch component-specific monitoring configuration
+	for _, componentFn := range []func() (string, string, error){
+		b.Shoot.Components.ControlPlane.KubeScheduler.MonitoringConfiguration,
```

Comment:
Can we rather have a clear interface for components, that want to contribute to the monitoring configuration?

Maybe the interface could then have 4 functions similar to the keys of the configmaps that extensions use to contribute config:
```go
type MonitoringComponent interface {
     ScrapeConfig() (string, error)
     AlertConfig() (string, error)
     DashboardsOperators() (string, error)
     DashboardsUsers() (string, error)
}
```
Or something similar. Maybe some return types could rather be `map[string]string` so we can then use `utils.MergeStringMaps()` instead of writing YAML formatted strings to string builders.

Maybe this also makes sense for the Logging part.
WDYT?